sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java

```diff
@@ -279,11 +279,17 @@ public Mono<StoreResponse> invokeStoreAsync(final Uri addressUri, final RxDocume
 
         final RntbdRequestArgs requestArgs = new RntbdRequestArgs(request, addressUri);
 
+        final boolean isAddressUriUnderOpenConnectionsFlow = this.proactiveOpenConnectionsProcessor
```

Comment:
this logic probably can be refactored into endpointProvider.createIfAbsent (no blocking, just saw this logic few times in different places)

Reply:
Will put this as a part of the refactoring PR.